src/main/java/seedu/address/ui/UiManager.java

```diff
@@ -134,4 +135,14 @@ public void handleLoginAppRequestEvent(LoginAppRequestEvent event) {
             Platform.runLater(() -> mainWindow.fillInnerParts());
         }
     }
+
+    /**
+     * Handles change internal list event.
+     * Displays the list that user requested(e.g mainlist, blacklist etc)
+     */
+    @Subscribe
+    private void handleChangeInternalListEvent(ChangeInternalListEvent event) {
+        logger.info(LogsCenter.getEventHandlingLogMessage(event));
+        mainWindow.fillInnerPartsWithIndicatedList(event.getListName());
```

Comment:
I think the problem you mentioned (list coming out before login) is here. In the original code, the inner parts are filled at login. The shortest way out I can think of is to set a variable, that checks whether the user is logged in, that can be accessed either here or in MainWindow.java where this method is called.
Jel made a getLoginStatus() in main/java/seedu/address/commons/events/ui/LoginAppRequest.java, that might help

Reply:
Noted. Thanks for the suggestion wei ping!!